scripts/reindex_articles.coffee

```diff
@@ -5,32 +5,115 @@ path = require 'path'
 Article = require '../src/api/apps/articles/model/index.js'
 ArticleModel = require '../src/api/models/article.coffee'
 search = require '../src/api/lib/elasticsearch'
-async = require 'async'
+asyncLib = require 'async'
 { cloneDeep } = require 'lodash'
 
-# Setup environment variables
 env = require 'node-env-file'
 switch process.env.NODE_ENV
   when 'test' then env path.resolve __dirname, '../.env.test'
   when 'production', 'staging' then ''
   else env path.resolve __dirname, '../.env'
 
-# Connect to database
-client = new MongoClient(process.env.MONGOHQ_URL)
-await client.connect()
-db = await client.db()
-articles = await db.collection('articles')
+console.log('Starting article reindexing process...')
+console.log('Environment:', process.env.NODE_ENV || 'development')
+console.log('MongoDB URL:', process.env.MONGOHQ_URL ? 'Configured' : 'NOT CONFIGURED')
 
-articles.find({}).toArray (err, articles) ->
-  console.log(err) if err
-  console.log(articles.length)
-  async.mapSeries(articles, indexWorker, (err, results) =>
-    console.log('Completed indexing ' + results.length + ' articles.')
-  )
+# Log Elasticsearch configuration
+console.log('Elasticsearch Configuration:')
+console.log('  - URL:', process.env.ELASTICSEARCH_URL || process.env.OPENSEARCH_URL || 'NOT CONFIGURED')
+console.log('  - Index Name:', search.index)
+console.log('  - Index Suffix:', process.env.ELASTICSEARCH_INDEX_SUFFIX || 'production')
+console.log('  - Using OpenSearch:', search.isOpenSearch ? 'Yes' : 'No')
+
+main = ->
```

Comment:
The multiple `.then` chaining makes this a bit tough to read. Unless it isn't supported for some reason, may be easier to read if we used `async/await`.

Reply:
I’m also curious how this script worked before, since async/await support was only introduced in CoffeeScript 2, and our current version is `"coffeescript": "1.11.1"`?